synapse/handlers/message.py

```diff
@@ -611,8 +611,13 @@ def handle_new_client_event(
             extra_users (list(UserID)): Any extra users to notify about event
         """
 
+        if event.is_state() and (event.type, event.state_key) == (EventTypes.Create, ""):
+            room_version = event.content["room_version"]
```

Comment:
I reckon you can get here without there being a `room_version`, if you hit `PUT /rooms/<room_id>/state/m.room.create`. Obviously that's a silly thing to do, but blowing up with an exception doesn't seem like the right thing to do.